Sources/StreamChat/Controllers/ListDatabaseObserver.swift

```diff
@@ -272,11 +272,25 @@ class ListChangeAggregator<DTO: NSManagedObject, Item>: NSObject, NSFetchedResul
             }
             currentChanges.append(.move(item, fromIndex: fromIndex, toIndex: toIndex))
             
+            // If an update already exists, remove it. A move acts as an update too.
+            currentChanges.removeAll(where: { change in
+                guard case let .update(_, index) = change else { return false }
+                return toIndex == index
+            })
+            
         case .update:
             guard let index = indexPath else {
                 log.warning("Skipping the update from DB because `indexPath` is missing for `.update` change.")
                 return
             }
+            
+            // If a move already exists, do not append an update change
+            let updateAlreadyExists = currentChanges.contains(where: { change -> Bool in
+                guard case let .move(_, _, indexPath) = change else { return false }
+                return index == indexPath
+            })
+            guard !updateAlreadyExists else { return }
+            
```

Comment:
Instead of doing it twice, wouldn't it be easier to do it once as the final step before publishing the changes?